PWGHF/TableProducer/treeCreatorXicToXiPiPi.cxx

```diff
@@ -0,0 +1,758 @@
+                  collision::PosZ);
+
+DECLARE_SOA_TABLE(HfCandXicFullPs, "AOD", "HFCANDXICFULLP",
+                  collision::BCId,
+                  full::Pt,
+                  full::Eta,
+                  full::Phi,
+                  full::Y,
+                  hf_cand_xic_to_xi_pi_pi::FlagMcMatchGen);
+} // namespace o2::aod
+
+/// Writes the full information in an output TTree
+struct HfTreeCreatorXicToXiPiPi {
+  Produces<o2::aod::HfCandXicLites> rowCandidateLite;
+  Produces<o2::aod::HfCandXicLiteKfs> rowCandidateLiteKf;
+  Produces<o2::aod::HfCandXicFulls> rowCandidateFull;
+  Produces<o2::aod::HfCandXicFullKfs> rowCandidateFullKf;
+  Produces<o2::aod::HfCandXicFullEvs> rowCandidateFullEvents;
+  Produces<o2::aod::HfCandXicFullPs> rowCandidateFullParticles;
+
+  Configurable<int> selectionFlagXic{"selectionXic", 1, "Selection Flag for Xic"};
+  Configurable<bool> fillCandidateLiteTable{"fillCandidateLiteTable", false, "Switch to fill lite table with candidate properties"};
+  // parameters for production of training samples
+  Configurable<bool> fillOnlySignal{"fillOnlySignal", false, "Flag to fill derived tables with signal for ML trainings"};
+  Configurable<bool> fillOnlyBackground{"fillOnlyBackground", false, "Flag to fill derived tables with background for ML trainings"};
+  Configurable<float> downSampleBkgFactor{"downSampleBkgFactor", 1., "Fraction of background candidates to keep for ML trainings"};
+  Configurable<float> ptMaxForDownSample{"ptMaxForDownSample", 10., "Maximum pt for the application of the downsampling factor"};
+
+  using SelectedCandidates = soa::Filtered<soa::Join<aod::HfCandXic, aod::HfSelXicToXiPiPi>>;
+  using SelectedCandidatesKf = soa::Filtered<soa::Join<aod::HfCandXic, aod::HfCandXicKF, aod::HfSelXicToXiPiPi>>;
+  using SelectedCandidatesMc = soa::Filtered<soa::Join<aod::HfCandXic, aod::HfCandXicMcRec, aod::HfSelXicToXiPiPi>>;
+  using SelectedCandidatesKfMc = soa::Filtered<soa::Join<aod::HfCandXic, aod::HfCandXicKF, aod::HfCandXicMcRec, aod::HfSelXicToXiPiPi>>;
+  using TracksWPid = soa::Join<aod::Tracks, aod::TracksPidPi>;
+
+  Filter filterSelectCandidates = aod::hf_sel_candidate_xic::isSelXicToXiPiPi >= selectionFlagXic;
+
+  Partition<SelectedCandidatesMc> recSig = nabs(aod::hf_cand_xic_to_xi_pi_pi::flagMcMatchRec) != int8_t(0);
+  Partition<SelectedCandidatesMc> recBg = nabs(aod::hf_cand_xic_to_xi_pi_pi::flagMcMatchRec) == int8_t(0);
+  Partition<SelectedCandidatesKfMc> recSigKf = nabs(aod::hf_cand_xic_to_xi_pi_pi::flagMcMatchRec) != int8_t(0);
+  Partition<SelectedCandidatesKfMc> recBgKf = nabs(aod::hf_cand_xic_to_xi_pi_pi::flagMcMatchRec) == int8_t(0);
+
+  void init(InitContext const&)
+  {
+  }
+
+  template <typename T>
+  void fillEvent(const T& collision)
+  {
+    rowCandidateFullEvents(
+      collision.bcId(),
+      collision.numContrib(),
+      collision.posX(),
+      collision.posY(),
+      collision.posZ());
+  }
+
+  template <bool doMc = false, bool noKf = true, typename T>
```

Comment:
```suggestion
  template <bool doMc, bool doKf, typename T>
```
It's more transparent in the code when template arguments are always passed.
`noKf` is a negation of `Kf`, which you don't use here, so it's better to keep it simple.